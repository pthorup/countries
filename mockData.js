const answer = 'Sweden'

const continent = {
   name: 'Africa',
   countries: [
      {
         __typename: 'Country',
         name: 'Angola',
         code: 'AO',
      },
      {
         __typename: 'Country',
         name: 'Burkina Faso',
         code: 'BF',
      },
      {
         __typename: 'Country',
         name: 'Burundi',
         code: 'BI',
      },
      {
         __typename: 'Country',
         name: 'Benin',
         code: 'BJ',
      },
      {
         __typename: 'Country',
         name: 'Botswana',
         code: 'BW',
      },
      {
         __typename: 'Country',
         name: 'Democratic Republic of the Congo',
         code: 'CD',
      },
      {
         __typename: 'Country',
         name: 'Central African Republic',
         code: 'CF',
      },
      {
         __typename: 'Country',
         name: 'Republic of the Congo',
         code: 'CG',
      },
      {
         __typename: 'Country',
         name: 'Ivory Coast',
         code: 'CI',
      },
      {
         __typename: 'Country',
         name: 'Cameroon',
         code: 'CM',
      },
      {
         __typename: 'Country',
         name: 'Cape Verde',
         code: 'CV',
      },
      {
         __typename: 'Country',
         name: 'Djibouti',
         code: 'DJ',
      },
      {
         __typename: 'Country',
         name: 'Algeria',
         code: 'DZ',
      },
      {
         __typename: 'Country',
         name: 'Egypt',
         code: 'EG',
      },
      {
         __typename: 'Country',
         name: 'Western Sahara',
         code: 'EH',
      },
      {
         __typename: 'Country',
         name: 'Eritrea',
         code: 'ER',
      },
      {
         __typename: 'Country',
         name: 'Ethiopia',
         code: 'ET',
      },
      {
         __typename: 'Country',
         name: 'Gabon',
         code: 'GA',
      },
      {
         __typename: 'Country',
         name: 'Ghana',
         code: 'GH',
      },
      {
         __typename: 'Country',
         name: 'Gambia',
         code: 'GM',
      },
      {
         __typename: 'Country',
         name: 'Guinea',
         code: 'GN',
      },
      {
         __typename: 'Country',
         name: 'Equatorial Guinea',
         code: 'GQ',
      },
      {
         __typename: 'Country',
         name: 'Guinea-Bissau',
         code: 'GW',
      },
      {
         __typename: 'Country',
         name: 'Kenya',
         code: 'KE',
      },
      {
         __typename: 'Country',
         name: 'Comoros',
         code: 'KM',
      },
      {
         __typename: 'Country',
         name: 'Liberia',
         code: 'LR',
      },
      {
         __typename: 'Country',
         name: 'Lesotho',
         code: 'LS',
      },
      {
         __typename: 'Country',
         name: 'Libya',
         code: 'LY',
      },
      {
         __typename: 'Country',
         name: 'Morocco',
         code: 'MA',
      },
      {
         __typename: 'Country',
         name: 'Madagascar',
         code: 'MG',
      },
      {
         __typename: 'Country',
         name: 'Mali',
         code: 'ML',
      },
      {
         __typename: 'Country',
         name: 'Mauritania',
         code: 'MR',
      },
      {
         __typename: 'Country',
         name: 'Mauritius',
         code: 'MU',
      },
      {
         __typename: 'Country',
         name: 'Malawi',
         code: 'MW',
      },
      {
         __typename: 'Country',
         name: 'Mozambique',
         code: 'MZ',
      },
      {
         __typename: 'Country',
         name: 'Namibia',
         code: 'NA',
      },
      {
         __typename: 'Country',
         name: 'Niger',
         code: 'NE',
      },
      {
         __typename: 'Country',
         name: 'Nigeria',
         code: 'NG',
      },
      {
         __typename: 'Country',
         name: 'Réunion',
         code: 'RE',
      },
      {
         __typename: 'Country',
         name: 'Rwanda',
         code: 'RW',
      },
      {
         __typename: 'Country',
         name: 'Seychelles',
         code: 'SC',
      },
      {
         __typename: 'Country',
         name: 'Sudan',
         code: 'SD',
      },
      {
         __typename: 'Country',
         name: 'Saint Helena',
         code: 'SH',
      },
      {
         __typename: 'Country',
         name: 'Sierra Leone',
         code: 'SL',
      },
      {
         __typename: 'Country',
         name: 'Senegal',
         code: 'SN',
      },
      {
         __typename: 'Country',
         name: 'Somalia',
         code: 'SO',
      },
      {
         __typename: 'Country',
         name: 'South Sudan',
         code: 'SS',
      },
      {
         __typename: 'Country',
         name: 'São Tomé and Príncipe',
         code: 'ST',
      },
      {
         __typename: 'Country',
         name: 'Swaziland',
         code: 'SZ',
      },
      {
         __typename: 'Country',
         name: 'Chad',
         code: 'TD',
      },
      {
         __typename: 'Country',
         name: 'Togo',
         code: 'TG',
      },
      {
         __typename: 'Country',
         name: 'Tunisia',
         code: 'TN',
      },
      {
         __typename: 'Country',
         name: 'Tanzania',
         code: 'TZ',
      },
      {
         __typename: 'Country',
         name: 'Uganda',
         code: 'UG',
      },
      {
         __typename: 'Country',
         name: 'Mayotte',
         code: 'YT',
      },
      {
         __typename: 'Country',
         name: 'South Africa',
         code: 'ZA',
      },
      {
         __typename: 'Country',
         name: 'Zambia',
         code: 'ZM',
      },
      {
         __typename: 'Country',
         name: 'Zimbabwe',
         code: 'ZW',
      },
   ],
}

module.exports = {
   answer,
   continent,
}
