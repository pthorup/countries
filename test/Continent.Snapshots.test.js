import React from 'react'
import Continent from '../pages/continent/[code]'
import renderer from 'react-test-renderer'
import { continent } from '../mockData'

it('All countries within a continent', () => {
   const tree = renderer.create(<Continent continent={continent} />)

   expect(tree).toMatchSnapshot()
})
