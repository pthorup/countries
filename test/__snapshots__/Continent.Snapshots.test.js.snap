// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All countries within a continent 1`] = `
<div>
  <h1>
    Africa
  </h1>
  <div
    className="countryList"
  >
    <div
      className="listItem"
    >
      <a
        href="/country/AO"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Angola
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/BF"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Burkina Faso
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/BI"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Burundi
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/BJ"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Benin
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/BW"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Botswana
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/CD"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Democratic Republic of the Congo
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/CF"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Central African Republic
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/CG"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Republic of the Congo
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/CI"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Ivory Coast
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/CM"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Cameroon
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/CV"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Cape Verde
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/DJ"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Djibouti
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/DZ"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Algeria
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/EG"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Egypt
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/EH"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Western Sahara
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/ER"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Eritrea
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/ET"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Ethiopia
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/GA"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Gabon
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/GH"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Ghana
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/GM"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Gambia
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/GN"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Guinea
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/GQ"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Equatorial Guinea
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/GW"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Guinea-Bissau
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/KE"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Kenya
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/KM"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Comoros
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/LR"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Liberia
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/LS"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Lesotho
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/LY"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Libya
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/MA"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Morocco
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/MG"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Madagascar
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/ML"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Mali
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/MR"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Mauritania
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/MU"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Mauritius
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/MW"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Malawi
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/MZ"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Mozambique
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/NA"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Namibia
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/NE"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Niger
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/NG"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Nigeria
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/RE"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Réunion
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/RW"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Rwanda
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/SC"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Seychelles
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/SD"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Sudan
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/SH"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Saint Helena
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/SL"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Sierra Leone
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/SN"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Senegal
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/SO"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Somalia
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/SS"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        South Sudan
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/ST"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        São Tomé and Príncipe
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/SZ"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Swaziland
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/TD"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Chad
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/TG"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Togo
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/TN"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Tunisia
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/TZ"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Tanzania
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/UG"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Uganda
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/YT"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Mayotte
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/ZA"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        South Africa
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/ZM"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Zambia
      </a>
    </div>
    <div
      className="listItem"
    >
      <a
        href="/country/ZW"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Zimbabwe
      </a>
    </div>
  </div>
</div>
`;
